json_serialization_helpers.go

```diff
@@ -0,0 +1,46 @@
+package jsonserialization
+
+import (
+	"encoding/json"
+	"reflect"
+
+	absser "github.com/microsoft/kiota-abstractions-go/serialization"
+)
+
+// Unmarshal parses JSON-encoded data using a ParsableFactory and stores it in the value pointed to by model.
+func Unmarshal[T absser.Parsable](data []byte, model *T, parser absser.ParsableFactory) error {
+	jpn, err := NewJsonParseNode(data)
```

Comment:
instead of newing the parse node, I'd get the parse node from the factory.
https://github.com/microsoft/kiota-abstractions-go/blob/1cd0de2ecba38efac3d683ba324a4982cfe31f06/serialization/parse_node_factory_registry.go#L15

The reasoning being: ParseNodes implement a proxy design pattern to enable features like dirty tracking with the in memory backing store, and we most likely want to have identical behaviour between those helpers and people doing things with the client.